put startup local env back in the default load path

StefanKarpinski opened the pull request:
we want to remove it eventually but for now this was causing too
much trouble so we'll put it back in the load path

StefanKarpinski committed:
put current local env back in the default load path

we want to remove it eventually but for now this was causing too
much trouble so we'll put it back in the load path

KristofferC commented:
Note that even with this, the Pkg docs are wrong and people keep opening issues (https://github.com/JuliaLang/Pkg.jl/issues/418 etc) on the Pkg repo and ask on slack why things don't work as they are documented.

Is it too much to ask to have `CurrentEnv()` back until we have the complete replacement story ready?

StefanKarpinski committed:
final new lines for all text files

StefanKarpinski committed:
Base.current_env(): return nothing in deleted directory

Test this and that having `@` and `@@` in the LOAD_PATH does not
cause any problems when starting Julia in a deleted directory.

StefanKarpinski commented:
The new version puts `@@` in the default load path which acts like the old `CurrentEnv` did.
